From 7d92568ed609f0fef5814e71b6b7bd10925662b8 Mon Sep 17 00:00:00 2001
From: Huang Qi <huangqi3@xiaomi.com>
Date: Tue, 10 Jun 2025 16:43:07 +0800
Subject: [PATCH 78/80] fix: use nlink_t type for st_nlink in struct stat
 definition for NuttX

(backport <https://github.com/rust-lang/libc/pull/4483>)
(cherry picked from commit 8620c2c249b6f1b99f026c0382c80ef03f497efc)
---
 src/unix/nuttx/mod.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/unix/nuttx/mod.rs b/src/unix/nuttx/mod.rs
index de734f9e0..69732d845 100644
--- a/src/unix/nuttx/mod.rs
+++ b/src/unix/nuttx/mod.rs
@@ -32,7 +32,7 @@ s! {
         pub st_dev: dev_t,
         pub st_ino: ino_t,
         pub st_mode: mode_t,
-        pub st_nlink: u64,
+        pub st_nlink: nlink_t,
         pub st_uid: u32,
         pub st_gid: u32,
         pub st_rdev: dev_t,
-- 
2.43.0

